vernac/classes.mli

```diff
@@ -84,8 +84,19 @@ module Event : sig
     | NewInstance of instance
 end
 
-(* Observers are called whenever a class or an instance are declared *)
-val add_observer : (Event.t -> unit) -> unit
+(** Activated observers are called whenever a class or an instance are declared.
+
+    [register_observer] is to be called once per process for a given
+    string, unless [override] is [true]. The registered observer is not activated.
+
+    Activation state is part of the summary. It is up to the caller to
+    use libobject for persistence if desired.
+*)
+val register_observer : name:string -> ?override:bool -> (Event.t -> unit) -> unit
```

Comment:
For more safety, I think it would be better to return some opaque "handle" type that is just the underlying string, and replace the name parameter by that handle in the two functions below. But maybe it's overengineering, I don't know.

Reply:
changed